azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineMSIHelper.java

```diff
@@ -0,0 +1,385 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.compute.implementation;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.compute.OperatingSystemTypes;
+import com.microsoft.azure.management.compute.ResourceIdentityType;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.compute.VirtualMachineExtension;
+import com.microsoft.azure.management.compute.VirtualMachineIdentity;
+import com.microsoft.azure.management.graphrbac.BuiltInRole;
+import com.microsoft.azure.management.graphrbac.RoleAssignment;
+import com.microsoft.azure.management.graphrbac.ServicePrincipal;
+import com.microsoft.azure.management.graphrbac.implementation.GraphRbacManager;
+import com.microsoft.azure.management.resources.fluentcore.model.Indexable;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+import rx.Observable;
+import rx.functions.Func0;
+import rx.functions.Func1;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+/**
+ * Utility class to set MSI and MSI related resources for a virtual machine.
+ */
+class VirtualMachineMSIHelper {
```

Comment:
improper camel casing: should be VirtualMachineMsiHelper.
This also applies to a number of methods inside, which use "MSI" instead of "Msi" in their names